[BUG]: Identifier "__nanosleep" is undefined

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cuda-python/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Runtime Error

### Component

cuda.core

### Describe the bug

Error reported during post-release testing of cuda.core 0.6.0; see nvbug 5939707 for details.

### How to Reproduce

see nvbug 5939707

### Expected behavior

The following tests should PASS:
FAILED memory_ipc/test_event_ipc.py::TestEventIpc::test_main[DeviceMR] - cuda...
FAILED test_helpers.py::test_latchkernel - cuda.core._utils.cuda_utils.NVRTCE...
FAILED test_event.py::test_error_timing_incomplete - cuda.core._utils.cuda_ut...
FAILED test_event.py::test_timing_success - cuda.core._utils.cuda_utils.NVRTC...

### Operating System

_No response_

### nvidia-smi output

+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 575.66                 Driver Version: 575.66         CUDA Version: 12.9     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  Quadro GP100                   On  |   00000000:01:00.0 Off |                  Off |
| 30%   44C    P0             26W /  235W |       2MiB /  16384MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+

--- a/cuda_core/tests/memory_ipc/test_event_ipc.py
+++ b/cuda_core/tests/memory_ipc/test_event_ipc.py
@@ -15,6 +15,7 @@ CHILD_TIMEOUT_SEC = 30
 NBYTES = 64
 
 
+@pytest.mark.skipif(Device().compute_capability.major < 7, reason="__nanosleep is only available starting Volta (sm70)")
 class TestEventIpc:
     """Check the basic usage of IPC-enabled events with a latch kernel."""
 

--- a/cuda_core/tests/test_event.py
+++ b/cuda_core/tests/test_event.py
@@ -1,4 +1,4 @@
-# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-FileCopyrightText: Copyright (c) 2024-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
 
@@ -21,6 +21,7 @@ def test_event_init_disabled():
         cuda.core._event.Event()  # Ensure back door is locked.
 
 
+@pytest.mark.skipif(Device().compute_capability.major < 7, reason="__nanosleep is only available starting Volta (sm70)")
 def test_timing_success(init_cuda):
     options = EventOptions(enable_timing=True)
     device = Device()
@@ -117,6 +118,7 @@ def test_error_timing_recorded():
         event3 - event2
 
 
+@pytest.mark.skipif(Device().compute_capability.major < 7, reason="__nanosleep is only available starting Volta (sm70)")
 def test_error_timing_incomplete():
     device = Device()
     device.set_current()

--- a/cuda_core/tests/test_helpers.py
+++ b/cuda_core/tests/test_helpers.py
@@ -1,4 +1,4 @@
-# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-FileCopyrightText: Copyright (c) 2024-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 #
 # SPDX-License-Identifier: Apache-2.0
 
@@ -17,6 +17,7 @@ ENABLE_LOGGING = False  # Set True for test debugging and development
 NBYTES = 64
 
 
+@pytest.mark.skipif(Device().compute_capability.major < 7, reason="__nanosleep is only available starting Volta (sm70)")
 def test_latchkernel():
     """Test LatchKernel."""
     log = TimestampedLogger(enabled=ENABLE_LOGGING)

